### PR TITLE
[BugFix] use new parser in meta replay & restore

### LINKED ISSUE #7479: create routine load with comments will fail after FE restart
### Steps to reproduce the behavior (Required)
Use dbeaver to send the following sql to create a routine job, then restart current FE master.
```sql
--- xxx
 CREATE ROUTINE LOAD job ON agg_table
 COLUMNS TERMINATED BY ",",
 COLUMNS (user_id, web_url, pv)
 PROPERTIES
 (
   "desired_concurrent_number" = "6",
   "max_batch_interval" = "10",
   "max_error_number" = "1000"
 )
 FROM KAFKA
 (
   "kafka_broker_list" = "172.26.92.155:9092",
   "kafka_topic" = "test",
   "property.kafka_default_offsets" = "OFFSET_BEGINNING",
   "kafka_partitions" = "0"
 );
```

### Expected behavior (Required)
master failover success.

### Real behavior (Required)
new master exit with error msg: Invalid statement index: 1, size: 1

### StarRocks version (Required)
 - You can get the StarRocks version by executing SQL `select current_version()`
main

### PR DESCRIPTION
**Pull request by padmejin**
## What type of PR is this：
- [x] bug
- [ ] feature
- [ ] enhancement
- [ ] refactor
- [ ] others

## Which issues of this PR fixes ：

Fixes #7479 

## Problem Summary(Required) ：

The root cause is a compatibility issue between the old parser and the new one. 
When parsing multiple statements with comments, the old parser will get one statement while the new parser will get two, the first of which is an empty statement. Thus if we store the index from the result that is parsed from a new parser,  then try to get the statement using results that is parsed from a old parser, we'll get an index error.
I'm replacing the old parser with the new one to solve this partly. 
hangyuan has fix this by manually remove comments in https://github.com/StarRocks/starrocks/pull/7508, so I only need to use the new parser by now.

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:heart_eyes: **pass** : 5 / 6 (83.33%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/qe/OriginStatement.java|0|1|00.00%|
|:large_blue_circle:|com/starrocks/analysis/CreateRoutineLoadStmt.java|2|2|100.00%|
|:large_blue_circle:|com/starrocks/alter/RollupJobV2.java|3|3|100.00%|

File: fe/fe-core/src/main/java/com/starrocks/alter/RollupJobV2.java [hunk 1]
```diff
@@ -825,22 +821,20 @@ public void gsonPostProcess() throws IOException {
...
-        analyzer.setIgnoreCast();
         CreateMaterializedViewStmt stmt = null;
         try {
-            stmt = (CreateMaterializedViewStmt) SqlParserUtils.getStmt(parser, origStmt.idx);
+            List<StatementBase> stmts = com.starrocks.sql.parser.SqlParser.parse(
+                    origStmt.originStmt, SqlModeHelper.MODE_DEFAULT);
+            stmt = (CreateMaterializedViewStmt) stmts.get(origStmt.idx);
             stmt.setIsReplay(true);
-            stmt.analyze(analyzer);
+            com.starrocks.sql.analyzer.Analyzer.analyze(stmt, connectContext);
```
[the remark this answers is not in the document]
Comment (padmejin) [reply fe/fe-core/src/main/java/com/starrocks/alter/RollupJobV2.java]: Done

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:disappointed: **fail** : 2 / 3 (66.67%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/qe/OriginStatement.java|0|1|00.00%|
|:large_blue_circle:|com/starrocks/analysis/CreateRoutineLoadStmt.java|2|2|100.00%|

File: fe/fe-core/src/main/java/com/starrocks/analysis/CreateRoutineLoadStmt.java [hunk 2]
```diff
@@ -306,9 +302,10 @@ public static RoutineLoadDesc getLoadDesc(OriginStatement origStmt, Map<String,
         }
 
         // parse the origin stmt to get routine load desc
-        SqlParser parser = new SqlParser(new SqlScanner(new StringReader(origStmt.originStmt), sqlMode));
         try {
-            StatementBase stmt = SqlParserUtils.getStmt(parser, origStmt.idx);
+            List <StatementBase> stmts = com.starrocks.sql.parser.SqlParser.parse(
```
Comment (gengjun-git) [inline fe/fe-core/src/main/java/com/starrocks/analysis/CreateRoutineLoadStmt.java:306]: If the original stmt stored in image is parsed by old parser. The index will be 0 even though there is comment before the sql. In this case, using the new parser will get the comment line.

Comment (padmejin) [reply fe/fe-core/src/main/java/com/starrocks/analysis/CreateRoutineLoadStmt.java:306]: Hangyuan has fix this by https://github.com/StarRocks/starrocks/pull/7508

**Review by gengjun-git — APPROVED** [review]

**Review by chaoyli — APPROVED** [review]

**Comment by padmejin** [comment]
@mergify backport branch-2.3 branch-2.2

**Pull request merged**